src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -258,7 +259,18 @@ private Task ProcessStreamItem(HubConnectionContext connection, StreamItemMessag
             else
             {
                 bool isStreamCall = descriptor.StreamingParameters != null;
-                return Invoke(descriptor, connection, hubMethodInvocationMessage, isStreamResponse, isStreamCall);
+                if (connection.ActiveInvocationLimit != null && !isStreamCall && !isStreamResponse)
+                {
+                    return connection.ActiveInvocationLimit.RunAsync(state =>
+                    {
+                        var (dispatcher, descriptor, connection, invocationMessage) = state;
+                        return dispatcher.Invoke(descriptor, connection, invocationMessage, isStreamResponse: false, isStreamCall: false);
```

Comment:
Can the above call to Invoke ever throw?

Reply:
Not really. I think if you wrote a custom auth handler and threw in there it could throw. We could wrap that so it closes the connection with an error, but would prefer it in a different change.